docs/source/references/config.md

```diff
@@ -0,0 +1,543 @@
+    "payload": ""
+  },
+  "KURA_CONFIGURATION": {
+    "KURA_INIT_MODE": "strict",
+    "KURA_BLOCK_STORE_PATH": "./blocks"
+  },
+  "SUMERAGI_CONFIGURATION": {
+    "PEER_ID": {
+      "address": "",
+      "public_key": "ed0100"
+    },
+    "BLOCK_TIME_MS": 1000,
+    "TRUSTED_PEERS": [],
+    "MAX_FAULTY_PEERS": 0,
+    "COMMIT_TIME_MS": 1000,
+    "TX_RECEIPT_TIME_MS": 200,
+    "N_TOPOLOGY_SHIFTS_BEFORE_RESHUFFLE": 1,
+    "MAX_INSTRUCTION_NUMBER": 4096
+  },
+  "TORII_CONFIGURATION": {
+    "TORII_P2P_URL": "127.0.0.1:1337",
+    "TORII_API_URL": "127.0.0.1:8080",
+    "TORII_MAX_TRANSACTION_SIZE": 32768,
+    "TORII_MAX_SUMERAGI_MESSAGE_SIZE": 16384000,
+    "TORII_MAX_INSTRUCTION_NUMBER": 4096
+  },
+  "BLOCK_SYNC_CONFIGURATION": {
+    "GOSSIP_PERIOD_MS": 10000,
+    "BATCH_SIZE": 4
+  },
+  "QUEUE_CONFIGURATION": {
+    "MAXIMUM_TRANSACTIONS_IN_BLOCK": 8192,
+    "MAXIMUM_TRANSACTIONS_IN_QUEUE": 65536,
+    "TRANSACTION_TIME_TO_LIVE_MS": 86400000
+  },
+  "LOGGER_CONFIGURATION": {
+    "MAX_LOG_LEVEL": "INFO",
+    "TERMINAL_COLOR_ENABLED": false,
+    "DATE_TIME_FORMAT": "%Y-%m-%d %H:%M:%S:%f"
+  },
+  "GENESIS_CONFIGURATION": {
+    "GENESIS_ACCOUNT_PUBLIC_KEY": "ed0100",
+    "GENESIS_ACCOUNT_PRIVATE_KEY": null,
+    "GENESIS_BLOCK_PATH": null,
+    "WAIT_FOR_PEERS_RETRY_COUNT": 0,
+    "WAIT_FOR_PEERS_RETRY_PERIOD_MS": 0
+  },
+  "WSV_CONFIGURATION": {
+    "ASSET_METADATA_LIMITS": {
+      "max_len": 1048576,
+      "max_entry_byte_size": 4096
+    },
+    "ACCOUNT_METADATA_LIMITS": {
+      "max_len": 1048576,
+      "max_entry_byte_size": 4096
+    }
+  }
+}
+```
+
+## `block_sync_configuration`
+
+`BlockSynchronizer` configuration.
+
+Has type `BlockSyncConfiguration`. Can be configured via environment variable `IROHA_BLOCK_SYNC_CONFIGURATION`
+
+```json
+{
+  "BATCH_SIZE": 4,
+  "GOSSIP_PERIOD_MS": 10000
+}
+```
+
+### `block_sync_configuration.batch_size`
+
+The number of blocks, which can be send in one message.
+
+Has type `u64`. Can be configured via environment variable `BLOCK_SYNC_BATCH_SIZE`
+
+```json
+4
+```
+
+### `block_sync_configuration.gossip_period_ms`
+
+The time between peer sharing its latest block hash with other peers in milliseconds.
+
+Has type `u64`. Can be configured via environment variable `BLOCK_SYNC_GOSSIP_PERIOD_MS`
+
+```json
+10000
+```
+
+## `genesis_configuration`
+
+Configuration for `GenesisBlock`.
+
+Has type `GenesisConfiguration`. Can be configured via environment variable `IROHA_GENESIS_CONFIGURATION`
+
+```json
+{
+  "GENESIS_ACCOUNT_PRIVATE_KEY": null,
+  "GENESIS_ACCOUNT_PUBLIC_KEY": "ed0100",
+  "GENESIS_BLOCK_PATH": null,
+  "WAIT_FOR_PEERS_RETRY_COUNT": 0,
+  "WAIT_FOR_PEERS_RETRY_PERIOD_MS": 0
+}
+```
+
+### `genesis_configuration.genesis_account_private_key`
+
+Genesis account private key, only needed on the peer that submits the genesis block.
+
+Has type `Option < PrivateKey >`. Can be configured via environment variable `IROHA_GENESIS_ACCOUNT_PRIVATE_KEY`
+
+```json
+null
```

Comment:
Ideally examples should cover all possible variants.. at least not None for Option. Is it issue related to defaults?

Reply:
Yes